src/gc.c

```diff
@@ -1109,6 +1138,13 @@ void jl_gc_count_allocd(size_t sz) JL_NOTSAFEPOINT
     jl_ptls_t ptls = jl_current_task->ptls;
     jl_atomic_store_relaxed(&ptls->gc_num.allocd,
         jl_atomic_load_relaxed(&ptls->gc_num.allocd) + sz);
+        uint64_t alloc_acc = jl_atomic_load_relaxed(&ptls->gc_num.alloc_acc);
+    if (alloc_acc + sz < 16*1024)
+        jl_atomic_store_relaxed(&ptls->gc_num.alloc_acc, alloc_acc + sz);
```

Comment:
Similary here, you could lose updates if you don't do atomic_inc

Reply:
alloc_acc is a per thread value, the global update is a couple lines below with the full `fetch_add`